src/pages/ParticipantsPage.js

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import {
+    View,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import styles from '../styles/styles';
+import ONYXKEYS from '../ONYXKEYS';
+import HeaderWithCloseButton from '../components/HeaderWithCloseButton';
+import Navigation from '../libs/Navigation/Navigation';
+import ScreenWrapper from '../components/ScreenWrapper';
+import OptionsList from '../components/OptionsList';
+import ROUTES from '../ROUTES';
+import CONST from '../CONST';
+
+const matchType = PropTypes.shape({
+    params: PropTypes.shape({
+        // reportID passed via route /participants/:reportID
+        reportID: PropTypes.string,
+    }),
+});
+
+const myPersonalDetailsType = PropTypes.shape({
+    // Display name of the current user from their personal details
+    displayName: PropTypes.string,
+
+    // Avatar URL of the current user from their personal details
+    avatar: PropTypes.string,
+
+    // login of the current user from their personal details
+    login: PropTypes.string,
+});
+
+const propTypes = {
+    /* Onyx Props */
+    // The personal details of the person who is logged in
+    myPersonalDetails: myPersonalDetailsType.isRequired,
+
+    // List of reports
+    reports: PropTypes.shape({
+        icons: PropTypes.arrayOf(PropTypes.string),
+        reportName: PropTypes.string,
+        participants: PropTypes.arrayOf(PropTypes.string),
+    }).isRequired,
+
+    // Route params
+    route: matchType.isRequired,
+
+    priorityMode: PropTypes.string.isRequired,
+};
+
+const getAllParticipants = (report, personalDetail) => {
+    const {icons, participants, reportName} = report;
+    const displayNames = reportName.split(', ');
+
+    const otherMembers = participants.map((login, index) => (
+        {
+            alternateText: login,
+            icons: [icons[index]],
+            keyForList: `${index}`,
+            tooltipText: login,
+            text: displayNames[index],
+            login,
+            displayName: displayNames[index],
+        }
+    ));
+
+    const final = [...otherMembers,
+        {
+            icons: [personalDetail.avatar],
+            text: personalDetail.displayName,
+            login: personalDetail.login,
+            alternateText: personalDetail.login,
+            keyForList: `${otherMembers.length}`,
+            tooltipText: personalDetail.login,
+            displayName: personalDetail.displayName,
+        },
+    ];
```

Comment:
Instead of making a bunch of temporary variables, couldn't we just start with the personalDetail object and then push the other members? Here's a loose version of what I'm thinking:
```
const members = [{
    alternateText: personalDetail.login,
    displayName: personalDetail.displayName,
    icons: [personalDetail.avatar],
    keyForList: `${participants.length}',
    login: personalDetail.login,
    text: personalDetail.displayName,
    tooltipText: personalDetail.login,
}];

participants.forEach((login, idx) => {
    members.push({
        alternateText: login,
        displayName: displayNames[idx],
        icons: [icons[idx]],
        keyForList: `${idx}',
        login,
        text: displayNames[idx],
        tooltipText: login,
    })
});
```

Even better could be to use a single reduce call:
```
const members = participants.reduce((list, login, idx) => {
    list.push({
        alternateText: login,
        displayName: displayNames[idx],
        icons: [icons[idx]],
        keyForList: `${idx}',
        login,
        text: displayNames[idx],
        tooltipText: login,
    });
    return list;
},{
    alternateText: personalDetail.login,
    displayName: personalDetail.displayName,
    icons: [personalDetail.avatar],
    keyForList: `${participants.length}',
    login: personalDetail.login,
    text: personalDetail.displayName,
    tooltipText: personalDetail.login,
});
```
I haven't ran either of these solutions, they're just recommendations. Also if we do end up keeping the code as is, it'd be nice to have the keys in the same order so that it's easier to tell they're the same type of object. Also renaming the `final` variable to be more semantically appropriate would be good (something like `allParticipants`)

Reply:
Hey, thanks for the suggestion, I did implement reduce alternative you proposed. It worked fine with one small tweak at the initialValue (should have been an array of objects rather than just the object).
The code is much cleaner, Thanks!